src/main/java/com/mixpanel/android/mpmetrics/MixpanelActivityLifecycleCallbacks.java

```diff
@@ -71,6 +71,10 @@ public void run() {
                             String sessionLengthString = nf.format((System.currentTimeMillis() - sStartSessionTime) / 1000);
                             JSONObject sessionProperties = new JSONObject();
                             sessionProperties.put(AutomaticEvents.SESSION_LENGTH, sessionLengthString);
+                            if (mMpInstance.getPeople().isIdentified()) {
```

Comment:
we don't need to check this. if the user is not identified, these updates will be queued in disk and flushed when `identify()` is called :)